src/shared/awsClientBuilder.ts

```diff
@@ -3,11 +3,21 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { Request, AWSError } from 'aws-sdk'
 import { ServiceConfigurationOptions } from 'aws-sdk/lib/service'
 import { env, version } from 'vscode'
 import { AwsContext } from './awsContext'
 import { extensionVersion } from './vscode/env'
 
+interface RequestExtras {
+    readonly service: AWS.Service
+    readonly operation: string
+    readonly params?: any
+}
+
+type RequestListener = (request: AWS.Request<any, AWSError> & RequestExtras) => void
+type ServiceOptions = ServiceConfigurationOptions & { requestListeners?: RequestListener[] }
```

Comment:
`onFoo` seems to be the convention in nodejs and we also use that in e.g. ChildProcess.
```suggestion
type ServiceOptions = ServiceConfigurationOptions & { onRequest?: RequestListener[] }
```